src/ostree/ot-builtin-commit.c

```diff
@@ -218,14 +218,18 @@ commit_editor (OstreeRepo     *repo,
   char **lines = NULL;
   int i;
 
-  *subject = NULL;
-  *body = NULL;
-
   input = g_strdup_printf ("\n"
       "# Please enter the commit message for your changes. The first line will\n"
       "# become the subject, and the remainder the body. Lines starting\n"
       "# with '#' will be ignored, and an empty message aborts the commit."
-      "%s%s\n", branch ? "\n#\n# Branch: " : "", branch ?: "");
+      "%s%s%s%s%s%s\n"
+              , branch ? "\n#\n# Branch: " : "", branch ? branch : ""
+              , *subject ? "\n" : "", *subject ? *subject : ""
+              , *body ? "\n" : "", *body ? *body : ""
+              );
+
+  *subject = NULL;
+  *body = NULL;
```

Comment:
``*subject`` has to be null so that the loop [assigns](https://github.com/Mathnerd314/ostree/blob/bbf94d5a26e24286dc2f644e8f661f09f97911d6/src/ostree/ot-builtin-commit.c#L254-L256) the first non-comment line to the subject. ``*body`` is null mostly because it was easier; it could also have been set to null in an else condition [here](https://github.com/Mathnerd314/ostree/blob/bbf94d5a26e24286dc2f644e8f661f09f97911d6/src/ostree/ot-builtin-commit.c#L281).